src/server/game/Spells/Spell.cpp

```diff
@@ -2262,6 +2262,19 @@ void Spell::AddDestTarget(SpellDestination const& dest, uint32 effIndex)
     m_destTargets[effIndex] = dest;
 }
 
+bool Spell::TargetInfo::MissedTarget(Spell *spell)
+{
+    if (spell->GetSpellInfo()->IsAffectingArea())
+        return false;
+
+    Unit* unit = spell->m_caster->GetGUID() == TargetGUID ? spell->m_caster->ToUnit() : ObjectAccessor::GetUnit(*spell->m_caster, TargetGUID);
+    if (!unit)
+        return true;
+    if (unit->HasStealthAura() && unit->IsHostileTo(spell->m_caster))
```

Comment:
https://github.com/TrinityCore/TrinityCore/pull/22971#discussion_r249412393 ?

Reply:
if (spell->GetSpellInfo()->IsAffectingArea())
        return false;

Should take care of ignoring the AoE's including Mass Dispel & Mind Sear or am I wrong?